float32 left
float32 right
int32 seq_num
